core/src/main/java/org/elasticsearch/action/support/replication/TransportReplicationAction.java

```diff
@@ -368,35 +392,50 @@ public void onFailure(Throwable e) {
 
         @Override
         protected void doRun() {
+            // shardID added to the request, as a side effect of calling resolveRequest
             if (checkBlocks() == false) {
```

Comment:
I renamed it to `checkBlocksAndResolveRequest` to clarify the ordering of the routines.